portfolio/src/main/webapp/script.js

```diff
@@ -39,21 +39,20 @@ function randQuote() {
 // Fetch comments from DataServlet.java
 function getComments() {
 
-  fetch('/comments').then(response => response.json()).then((comments) => {
+  fetch('/comments').then(response => response.json()).then((cmtContainer) => {
 
-    var str_comments = "<h2>Comments</h2><p>";
+    if(!cmtContainer) return;
 
-    for (var index in comments){
-      str_comments += comments[index] + "<br>";
+    var str_comments = "<h2>Comments</h2>";
+
+    for (var index in cmtContainer) {
```

Comment:
It looks to me like `cmtContainer` will be an array here (please correct me if I'm wrong!), in which case I would recommend using either [`forEach`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) or a [`for...of`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) loop.

Reply:
cmtContainer is an ArrayList in java but its converted to an object in java script.

Reply:
Technically arrays are sub-types of objects in JS (`typeof` will give you "object" on an array so you can use [`Array.isArray`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray) when you need to tell for sure). If this is a generic object though with integer properties (aka an "array-like object") then you can cast it to a true array using [`Array.from`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from).

If you use a `for...of` loop then change the `var` here to `const` as well.